great_expectations/util.py

```diff
@@ -1747,6 +1749,24 @@ def get_context(
     ...
 
 
+@overload
+def get_context(
+    project_config: Union[DataContextConfig, Mapping] = ...,  # non-null
+    context_root_dir: None = ...,  # null
+    runtime_environment: Optional[dict] = ...,
+    cloud_base_url: None = ...,
+    cloud_access_token: None = ...,
+    cloud_organization_id: None = ...,
+    cloud_mode: Optional[Literal[False]] = ...,
+    # <GX_RENAME> Deprecated as of 0.15.37
+    ge_cloud_base_url: None = ...,
+    ge_cloud_access_token: None = ...,
+    ge_cloud_organization_id: None = ...,
+    ge_cloud_mode: Optional[Literal[False]] = ...,
+) -> EphemeralDataContext:
+    ...
+
```

Comment:
Should still add the `CloudDataContext` here. Although I don't think it's worth it to enumerate all the possible permutations of args that could create a `CloudDataContext`.
Would that be `8!` (factorial)??

I do think we ignore the overlap error as long as the common cases are "revealing" the correct types. In another similar mypy issue, one of the mypy contributors even recommends an `ignore` comment as a solution.

https://github.com/python/mypy/issues/8656#issuecomment-1193170195

Reply:
```python
@overload
def get_context(
    ... # not sure what the other items can be
    cloud_base_url: str = ...,
    cloud_access_token: str = ...,
    cloud_organization_id: str = ...,
    cloud_mode: Optional[Literal[True]] = ...,
    # <GX_RENAME> Deprecated as of 0.15.37
    ge_cloud_base_url: Optional[str] = ...,
    ge_cloud_access_token: Optional[str] = ...,
    ge_cloud_organization_id: Optional[str] = ...,
    ge_cloud_mode: Optional[Literal[True]] = ...,
) -> CloudDataContext:
    ...

@overload
def get_context(
    ... # not sure what the other items can be
    cloud_base_url: Optional[str] = ...,
    cloud_access_token: Optional[str] = ...,
    cloud_organization_id: Optional[str] = ...,
    cloud_mode: Literal[True] = ...,
    # <GX_RENAME> Deprecated as of 0.15.37
    ge_cloud_base_url: Optional[str] = ...,
    ge_cloud_access_token: Optional[str] = ...,
    ge_cloud_organization_id: Optional[str] = ...,
    ge_cloud_mode: Optional[Literal[True]] = ...,
) -> CloudDataContext:
    ...

```